cpu/esp32/esp-now/esp_now_netdev.c

```diff
@@ -214,13 +214,16 @@ static IRAM_ATTR void esp_now_recv_cb(const uint8_t *mac, const uint8_t *data, i
     }
 #endif
 
-    if (_esp_now_dev.rx_pkt.len) {
-        /* there is already a packet in receive buffer, we drop the new one */
+    critical_enter();
+    mutex_lock(&_esp_now_dev.dev_lock);
+
+    if ((int)ringbuffer_get_free(&_esp_now_dev.rx_buf) < 1 + ESP_NOW_ADDR_LEN + len) {
```

Comment:
Maybe, add an comment for which purpose the extra byte is required and why 1 byte is sufficint to store the length of type int.